central/declarativeconfig/manager_impl.go

```diff
@@ -251,13 +278,18 @@ func (m *managerImpl) doUpsert(transformedMessagesByHandler map[string]protoMess
 			for _, message := range messages {
 				err := typeUpdater.Upsert(m.reconciliationCtx, message)
 				m.updateHealthForMessage(handler, message, err, consecutiveReconciliationErrorThreshold)
+				if err != nil {
+					failureInUpsert = true
```

Comment:
nit: `failureInUpsert = err != nil`

Reply:
Can't make the change like this, this will set `failureInUpsert` to false if the _last_ upsert didn't have any errors, although all previous ones did. So, we would indicate that "no error happened during upserts" which is wrong.